modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,590 @@
+            #    sum   pow2_sum  count
+            #    a  b  a  b      a  b
+            # a  1  2  3  4      5  6
+            # b  1  2  3  4      5  6
+            #
+            # And the 'parts_without_nans' like:
+            #    sum  pow2_sum  count
+            # a  1    3         5
+            # b  2    4         6
+            #
+            # Here we want to sum them in an order so the digit matches (1 + 1), (2 + 2), ...
+            # For that we first have to repeat the values in 'parts_without_nans':
+            #  parts_without_nans.repeat(parts_with_nans.shape[0]):
+            #    sum  pow2_sum  count
+            # a  1    3         5
+            # b  1    3         5
+            # a  2    4         6
+            # b  2    4         6
+            #
+            # And then reshape it using the "Fortran" order:
+            #  parts_without_nans.reshape(parts_with_nans.shape, order="F"):
+            #    sum   pow2_sum  count
+            #    a  b  a  b      a  b
+            # a  1  2  3  4      5  6
+            # b  1  2  3  4      5  6
+            # After that the shapes & orders are aligned and we can perform the summation
+            .repeat(repeats=len(parts_with_nans), axis=0).reshape(
+                parts_with_nans.shape, order="F"
+            )
+        )
+        replace_values = parts_with_nans + parts_without_nans
+
+        if not total_agg.values.flags.writeable:
+            # making a copy if the buffer is read-only as
+            # we will need to modify `total_agg` inplace
+            total_agg = total_agg.copy()
+        total_agg.values[:, na_agg_idxs] = replace_values
+
+        return total_agg
+
+    @staticmethod
+    def _build_corr_table_nan(
+        sum_of_pairwise_mul: pandas.DataFrame,
+        means: pandas.DataFrame,
+        sums: pandas.DataFrame,
+        count: pandas.DataFrame,
+        std: pandas.DataFrame,
+        cols: pandas.Index,
+        min_periods: int,
+    ) -> pandas.DataFrame:
+        """
+        Build correlation matrix for a DataFrame that had NaN values in it.
+
+        Parameters
+        ----------
+        sum_of_pairwise_mul : pandas.DataFrame
+        means : pandas.DataFrame
+        sums : pandas.DataFrame
+        count : pandas.DataFrame
+        std : pandas.DataFrame
+        cols : pandas.Index
+        min_periods : int
+
+        Returns
+        -------
+        pandas.DataFrame
+            Correlation matrix.
+        """
+        res = pandas.DataFrame(index=cols, columns=cols, dtype="float")
+
+        for col_idx, i in enumerate(cols):
+            mul_row = sum_of_pairwise_mul.loc[i]
+            mean_row = means.loc[i]
+            sum_row = sums.loc[i]
+            count_row = count.loc[i]
+            std_row = std.loc[i]
+
+            if count_row[i] >= min_periods:
+                res.loc[i, i] = 1.0
+
+            for j in cols[col_idx + 1 :]:
+                if count_row.loc[j] < min_periods:
```

Comment:
Here we can try to do something similar to https://github.com/modin-project/modin/pull/6193/files#r1213171254, and then fill the matrix with NaNs by `count<min_periods` mask.